src/screens/NotificationsPromoSheet/index.tsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import * as perms from 'react-native-permissions';
+
+import useAppState from '@/hooks/useAppState';
+import { useNavigation } from '@/navigation/Navigation';
+import { CampaignKey } from '@/campaigns/campaignChecks';
+import { PromoSheet } from '@/components/PromoSheet';
+import backgroundImage from '@/assets/notificationsPromoSheetBackground.png';
+import headerImageIOS from '@/assets/notificationsPromoSheetHeaderIOS.png';
+import headerImageAndroid from '@/assets/notificationsPromoSheetHeaderAndroid.png';
+import { delay } from '@/helpers/utilities';
+import Routes from '@/navigation/routesNames';
+import { useTheme } from '@/theme';
+import * as i18n from '@/languages';
+import { IS_IOS } from '@/env';
+import { logger, RainbowError } from '@/logger';
+
+const HEADER_HEIGHT = 255;
+const HEADER_WIDTH = 390;
+const TRANSLATIONS = i18n.l.promos.notifications_launch;
+
+export function NotificationsPromoSheetInner({
+  permissions,
+  requestNotificationPermissions,
+}: {
+  permissions: perms.NotificationsResponse;
+  requestNotificationPermissions: () => Promise<void>;
+}) {
+  const { colors } = useTheme();
+  const { goBack, navigate } = useNavigation();
+
+  const { status, settings } = permissions;
+  const hasSettingsEnabled = Boolean(
+    Object.values(settings).find(s => Boolean(s))
+  );
+  const notificationsEnabled = status === perms.RESULTS.GRANTED;
+  const notificationsDenied = status === perms.RESULTS.DENIED;
```

Comment:
maybe this would be cleaner as `switch (status) { case perms.RESULTS.GRANTED: ... }` etc